### PR TITLE
Param class in now replaced by In

### PR DESCRIPTION
**Pull request by mohammadpz**
Hi @nouiz ,
This PR is for solving #3395 .

### CODE REVIEW

File: doc/library/compile/function.txt [hunk 1]
```diff
@@ -50,7 +50,7 @@ Reference
 
         Initialize attributes from arguments.
 
-.. class:: Param
```
Comment (nouiz) [inline doc/library/compile/function.txt]: The list of attribute isn't in the same order and the In class have more attribute. Look at the docstring of the In class in theano/compile/io.py and its method `__init__`

File: doc/library/compile/function.txt [hunk 2]
```diff
@@ -27,7 +27,7 @@ array(8.0)
 The idea here is that we've compiled the symbolic graph (``2*x``) into a function that can be called on a number and will do some computations.
 
 The behaviour of function can be controlled in several ways, such as
-:class:`Param`, ``mode``, ``updates``, and ``givens``.  These are covered
+:class:`In`, ``mode``, ``updates``, and ``givens``.  These are covered
```
Comment (nouiz) [inline doc/library/compile/function.txt]: add the Out to the list and add the class Out after the class In from line 53.

**Comment by nouiz** [comment]
Other then the current comment, all seem good. thanks

File: doc/library/compile/function.txt (hunk 1 above)
Comment (mohammadpz) [inline doc/library/compile/function.txt]: For the new arguments, I wrote a shorter version of the description in ```__init__```. Could you please check if that is right?

**Comment by lamblin** [comment]
Thanks!

File: theano/__init__.py [hunk 3]
```diff
@@ -67,7 +67,7 @@
     predefined_modes, predefined_linkers, predefined_optimizers,
     FunctionMaker, function, function_dump, OpFromGraph,
     ProfileMode, ProfileStats,
-    Param, shared, as_op)
```
Comment (lamblin) [inline theano/__init__.py]: Actually, this is breaking something in the deep learning tutorials.
Should we keep an alias `Param = In`, or a dummy `Param` class that calls `In` with a deprecation warning?

Comment (nouiz) [inline theano/__init__.py]: What about making paran a sub class or in and adding the missing, changed
name parameter to init?

Le mar. 26 janv. 2016 14:34, Pascal Lamblin <notifications@github.com> a
écrit :

Comment (mohammadpz) [inline theano/__init__.py]: Please let me know if you make a decision. Thanks.

**Pull request merged**